.cbmc-batch/source/make_common_data_structures.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <openssl/ec.h>
+#include <openssl/evp.h>
+
+#include <cipher_openssl.h>
+#include <ec_utils.h>
+#include <evp_utils.h>
+#include <make_common_data_structures.h>
+
+void ensure_alg_properties_has_allocated_names(struct aws_cryptosdk_alg_properties *const alg_props) {
+    size_t md_name_size;
+    alg_props->md_name = can_fail_malloc(md_name_size);
+    size_t cipher_name_size;
+    alg_props->cipher_name = can_fail_malloc(cipher_name_size);
+    size_t alg_name_size;
+    alg_props->alg_name = can_fail_malloc(alg_name_size);
+}
+
+void ensure_md_context_has_allocated_members(struct aws_cryptosdk_md_context *ctx) {
+    ctx->alloc      = nondet_bool() ? NULL : can_fail_allocator();
+    ctx->evp_md_ctx = evp_md_ctx_nondet_alloc();
+}
+
+void ensure_sig_ctx_has_allocated_members(struct aws_cryptosdk_sig_ctx *ctx) {
+    ctx->alloc   = nondet_bool() ? NULL : can_fail_allocator();
+    ctx->keypair = ec_key_nondet_alloc();
+    ctx->pkey    = evp_pkey_nondet_alloc();
+    ctx->ctx     = evp_md_ctx_nondet_alloc();
+
+    // Need to ensure consistency of reference count later by assuming ctx is valid
+    evp_pkey_set0_ec_key(ctx->pkey, ctx->keypair);
```

Comment:
This is manually setting inside an ensure because the ordering of checks inside the is_valid (without short circuiting boolean evaluation) was accessing a member of a null pointer.  Lucas will look into this.

Reply:
Whats the status now?

Reply:
I need to ensure that `ctx->keypair->group` and `ctx->pkey->ec_key->group` are the same object, but for some reason I can't get it to work by just assuming `ctx->keypair == ctx->pkey->ec_key`. So I need to set `ctx->pkey->ec_key` to `ctx->keypair` manually.